src/test/string_utils_test.cpp

```diff
@@ -137,4 +137,19 @@ TEST_CASE("string_util tests") {
             CHECK(split_lines<vs>("first\r\n\r\nthird\r\n") == vs{"first", "", "third"});
         }
     }
+
+    SECTION("lower_case") {
+        SECTION("already lower") { CHECK(lower_case("already lowercase") == "already lowercase"); }
+        SECTION("all upper") { CHECK(lower_case("I AM SHOUTY") == "i am shouty"); }
+        SECTION("mixture") { CHECK(lower_case("I am A LITTLE shouty") == "i am a little shouty"); }
+        SECTION("non-letters") { CHECK(lower_case("G0RD0N'5 AL111111V3!!!!!!") == "g0rd0n'5 al111111v3!!!!!!"); }
+    }
+
+    SECTION("has_prefix") {
+        SECTION("prefix") { CHECK(has_prefix("boolean", "bool") == true); }
```

Comment:
I much prefer not `== false` or `==true` here: I find it visually noisy...

Also CHECK() is a magic macro that tries to understand and print out the intention of a (failing) test. It's smart enough to know if something was a bool check and prints something nice in that case. Instead of "expected 'true' got 'false'", which is...sort of redundant.

Minor gripe anyway! I know other languages are big on `== true` or `!= nullptr` but C++ idiomatically (IME!) isn't like that.

Reply:
Fair enough. I don't feel strongly either way, so switched to ditch the == true and == false.